gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/java/io/gravitee/am/gateway/handler/root/resources/endpoint/mfa/MFAChallengeEndpoint.java

```diff
@@ -281,6 +292,7 @@ private void verifyCode(RoutingContext routingContext) {
                                 if (verifyAttemptService.shouldSendEmail(client, domain)) {
                                     emailService.send(Template.VERIFY_ATTEMPT, endUser, client);
                                 }
+                                updateAuditLog(MFA_MAX_ATTEMPT_REACHED, endUser, client, factor, factorCtx, error);
                                 handleException(routingContext, VERIFY_ATTEMPT_ERROR_PARAM_KEY, "maximum_verify_limit");
                             } else {
                                 logger.error("Could not check verify attempts", error);
```

Comment:
missing log?

Reply:
log added

Reply:
I thought about the else block, there is only a logged error logger.error("Could not check verify attempts", error) but there will be nothing in Audit log